### PR TITLE
Flood checkboxes

### PR DESCRIPTION
**Pull request by p-chavoshi**
**https://trello.com/c/szzCrDpt/217-5-aauser-on-the-report-detail-view-below-the-common-field-section-for-flood-i-can-see-the-last-section-grouped-in-tables-with-ch**

image

### CODE REVIEW

File: apps/frontend/components/FormValidation/FloodFormValidation/FloodCheckBoxes.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            <Box key={int}>
+              <FormControlLabel
+                key={int}
+                control={
+                  <Checkbox
+                    disabled={!isEditable}
+                    onChange={changeFn}
+                    name={int.toString()}
+                    key={int}
+                    checked={threats?.split(' ').includes(key)}
```
Comment (sifelix) [inline apps/frontend/components/FormValidation/FloodFormValidation/FloodCheckBoxes.tsx:63]: use threatArr

File: apps/frontend/components/FormValidation/FloodFormValidation/FloodCheckBoxes.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    setThreatsArr(threatsCopy);
+  };
+
+  return (
+    <FormControl component="fieldset" sx={{ m: 3 }} variant="standard">
+      <FormGroup>
+        <FormLabel>
+          <FormattedMessage id="table.FLOOD.securityIssues.title" />
+        </FormLabel>
+        {range(1, 15).map(int => {
```
Comment (sifelix) [inline apps/frontend/components/FormValidation/FloodFormValidation/FloodCheckBoxes.tsx:50]: rename int key?

File: apps/frontend/components/FormValidation/FloodFormValidation/FloodTables.tsx [hunk 3]
```diff
@@ -14,7 +15,8 @@ export type FloodSpecificType = Record<
 
 interface IProps {
   value: FloodSpecificType;
-  onChange: (event: FloodSpecificType) => void;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  onChange: (event: any) => void;
```
Comment (sifelix) [inline apps/frontend/components/FormValidation/FloodFormValidation/FloodTables.tsx:19]: This change is not valid anymore

File: apps/frontend/components/FormValidation/FloodFormValidation/FloodTables.tsx [hunk 4]
```diff
@@ -32,6 +34,12 @@ export const FloodTables = ({
         onChange={onChange}
         isEditable={isEditMode}
       />
+      <FloodCheckBoxes
+        threats={value.threat}
```
Comment (sifelix) [inline apps/frontend/components/FormValidation/FloodFormValidation/FloodTables.tsx:38]: you already have value

File: packages/interfaces/package.json
Comment (sifelix) [inline packages/interfaces/package.json:1]: Why are you modifying interface package?

File: apps/frontend/components/FormValidation/FloodFormValidation/FloodCheckBoxes.tsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  FormControl,
+  FormControlLabel,
+  FormGroup,
+  FormLabel,
+} from '@mui/material';
+import { range, remove } from 'lodash';
+import { ChangeEvent, useState } from 'react';
+import { FormattedMessage, useIntl } from 'react-intl';
+
+import { FloodSpecificType } from './FloodTables';
```
Comment (sifelix) [inline apps/frontend/components/FormValidation/FloodFormValidation/FloodCheckBoxes.tsx:13]: don't import FloodSpecificType  from FloodTables

File: packages/interfaces/package.json
Comment (p-chavoshi) [reply packages/interfaces/package.json:1]: leftover from another change, will remove

**Review by sifelix — APPROVED** [review]

**Review by sifelix — APPROVED** [review]

**Pull request merged**